frappe/database.py

```diff
@@ -50,7 +50,7 @@ def connect(self):
 		"""Connects to a database as set in `site_config.json`."""
 		warnings.filterwarnings('ignore', category=MySQLdb.Warning)
 		self._conn = MySQLdb.connect(user=self.user, host=self.host, passwd=self.password,
-			use_unicode=True, charset='utf8mb4')
+			use_unicode=True, charset='utf8mb4', read_default_file="/etc/mysql/my.cnf")
 		self._conn.converter[246]=float
```

Comment:
The standard way to handle this would be to add a new parameter in `site_config.json`. We can call it `my_cnf_path`

and then in the connect you can use `read_default_file = frappe.conf.get('my_cnf_path')`. This way we don't hard code anthing.